delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/ArchiveDownload.scala

```diff
@@ -0,0 +1,189 @@
+          case FileRejection.AuthorizationFailed => AuthorizationFailed(ref.ref, project)
+          case other                             => WrappedFileRejection(other)
+        }
+      if (ignoreNotFound) tarEntryIO.onErrorRecover { case _: ResourceNotFound => None }
+      else tarEntryIO
+    }
+
+    private def fetchResource(ref: ResourceReference, project: ProjectRef, ignoreNotFound: Boolean)(implicit
+        caller: Caller,
+        col: AclCollection
+    ): IO[ArchiveRejection, Option[(TarArchiveMetadata, Source[ByteString, NotUsed])]] = {
+      val refProject = ref.project.getOrElse(project)
+      if (hasPermission(resources.read, refProject)) {
+        val tarEntryIO = resourceRefToByteString(ref, project).map { content =>
+          val path     = pathOf(ref, project)
+          val metadata = TarArchiveMetadata.create(path, content.length.toLong)
+          Some((metadata, Source.single(content)))
+        }
+        if (ignoreNotFound) tarEntryIO.onErrorHandle { _: ResourceNotFound => None }
+        else tarEntryIO
+      } else IO.raiseError(AuthorizationFailed(ref.ref, refProject))
+    }
+
+    private def resourceRefToByteString(
+        ref: ResourceReference,
+        project: ProjectRef
+    ): IO[ResourceNotFound, ByteString] = {
+      val p = ref.project.getOrElse(project)
+      val r = ref.representation.getOrElse(CompactedJsonLd)
+      Stream
+        .fromIterator[Task](exchanges.iterator)
+        .evalMap { _(p, ref.ref) }
+        .collectFirst { case Some(value) => value }
+        .evalMap { value => valueToByteString(value, r) }
+        .compile
+        .toList
+        .hideErrors
+        .flatMap { list =>
+          list.headOption match {
+            case Some(value) => UIO.pure(value)
+            case None        => IO.raiseError(ResourceNotFound(ref.ref, project))
+          }
+        }
+    }
+
+    private def valueToByteString(
+        value: ReferenceExchangeValue[_],
+        repr: ArchiveResourceRepresentation
+    ): IO[RdfError, ByteString] = {
+      repr match {
+        case SourceJson      => UIO.pure(value.toSource.sort.spaces2).map(v => ByteString(v))
+        case CompactedJsonLd => value.toCompacted.map(_.json.sort.spaces2).map(v => ByteString(v))
+        case ExpandedJsonLd  => value.toExpanded.map(_.json.spaces2).map(v => ByteString(v))
+        case NTriples        => value.toNTriples.map(v => ByteString(v.value))
+        case Dot             => value.toDot.map(v => ByteString(v.value))
+      }
+    }
+
+    private def pathOf(ref: ResourceReference, project: ProjectRef): String = {
+      val p = ref.project.getOrElse(project)
+      s"$p/${urlEncode(ref.ref.original)}.json"
+    }
+
+    private def pathOf(ref: FileReference, project: ProjectRef, filename: String): String = {
+      val p = ref.project.getOrElse(project)
+      s"$p/$filename"
+    }
+
+    private def hasPermission(
```

Comment:
no use, make use of `authorizeForAny` as explained before before

Reply:
please see my comment above